### PR TITLE
Adds Authentication

### PR DESCRIPTION
**Pull request by OmKakatkar**
## Features
- Auth Context
- [Login](https://deploy-preview-1--ani-watch.netlify.app/login) and [SignUp](https://deploy-preview-1--ani-watch.netlify.app/signup) form
- Constants folder with formdata, api endpoints
- Auth request functions
- Error handling with Toast Notification
- Route for mockman
- Changes for rendering as per React 18

### CODE REVIEW

**Review by thegeorgenikhil — COMMENTED** [review]
Can you check the deploy preview link? not working for me.
 **edit**: It was working fine.Didn't notice there wasn't any home routes defined.✔️

**Comment by OmKakatkar** [comment]
> Can you check the deploy preview link? not working for me.

Hi, you can click on the Login and SignUp links in the PR description.
I have removed the netlify bot comment as routing is not fully implemented

**Review by thegeorgenikhil — APPROVED** [review]
- Good way of making the auth proccess as functions.
- Learnt about using @params and @return
- The functionalities works fine and toast messages are good!

File: src/frontend/constants/api-constant.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Constants for Public api routes
+export const API_LOGIN = '/api/auth/login';
+export const API_SIGNUP = '/api/auth/signup';
+export const API_VIDEO = '/api/videos';
+export const API_CATEGORY = '/api/categories';
+
```
Comment (shrey27) [inline src/frontend/constants/api-constant.js:6]: it's good you have defined api endpoints all together

**Review by shrey27 — APPROVED** [review]
Code is clean, well structured and properly documented
UI looks good
PR is ready to merge

**Pull request merged**